test/e2e/framework/kubectl/kubectl_util.go

```diff
@@ -0,0 +1,292 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	utilversion "k8s.io/apimachinery/pkg/util/version"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	uexec "k8s.io/utils/exec"
+)
+
+const defaultHost = "http://127.0.0.1:8080"
+
+var (
+	// For parsing Kubectl version for version-skewed testing.
+	gitVersionRegexp = regexp.MustCompile("GitVersion:\"(v.+?)\"")
+)
+
+// KubectlBuilder is used to build, customize and execute a kubectl Command.
+// Add more functions to customize the builder as needed.
+type KubectlBuilder struct {
+	// Cmd is the custom command executed by KubectlBuilder.
+	Cmd     *exec.Cmd
+	timeout <-chan time.Time
+}
+
+// TestContext is a struct containing the minimum attributes needed to run KubectlCmd.
+type TestContext struct {
```

Comment:
Similar to your suggestion for a `NewKubeconfig` struct I've made a `NewTestContext` struct (the name made more sense to me).

Reply:
nit: could we name the struct something other than `TestContext`? Maybe `TestKubeconfig` something more descriptive?

Reply:
Yes, I can do that

Reply:
The reason why I named it `TestContext` is because it is a subset of the original `TestContext` and it contains `KubeConfig` so I didn't want that part to be ambiguous. @alejandrox1